crates/ruff_python_formatter/src/expression/parentheses.rs

```diff
@@ -110,26 +110,57 @@ where
 {
     FormatParenthesized {
         left,
+        comments: &[],
+        content: Argument::new(content),
+        right,
+    }
+}
+
+/// Formats `content` enclosed by the `left` and `right` parentheses, along with any dangling
+/// comments that on the parentheses themselves.
+pub(crate) fn parenthesized_with_dangling_comments<'content, 'ast, Content>(
+    left: &'static str,
+    comments: &'content [SourceComment],
+    content: &'content Content,
+    right: &'static str,
+) -> FormatParenthesized<'content, 'ast>
+where
+    Content: Format<PyFormatContext<'ast>>,
+{
+    FormatParenthesized {
+        left,
+        comments,
         content: Argument::new(content),
         right,
     }
 }
 
 pub(crate) struct FormatParenthesized<'content, 'ast> {
     left: &'static str,
+    comments: &'content [SourceComment],
     content: Argument<'content, PyFormatContext<'ast>>,
     right: &'static str,
 }
 
 impl<'ast> Format<PyFormatContext<'ast>> for FormatParenthesized<'_, 'ast> {
     fn fmt(&self, f: &mut Formatter<PyFormatContext<'ast>>) -> FormatResult<()> {
         let inner = format_with(|f| {
-            group(&format_args![
-                text(self.left),
-                &soft_block_indent(&Arguments::from(&self.content)),
-                text(self.right)
-            ])
-            .fmt(f)
+            if self.comments.is_empty() {
+                group(&format_args![
+                    text(self.left),
+                    &soft_block_indent(&Arguments::from(&self.content)),
+                    text(self.right)
+                ])
+                .fmt(f)
+            } else {
+                group(&format_args![
```

Comment:
Nit: Do we need this `group`? I believe it always expands because `dangling_comments` renders a hard line break or empty line.